src/schema-association-service.ts

```diff
@@ -237,11 +258,49 @@ async function autoDetectSchema(
     if (session === undefined) {
         logger.log(`No organization found for ${workspaceFolder.name}`, 'SchemaDetection');
         vscode.window.showErrorMessage(format(Messages.unableToAccessOrganization, organizationName));
+
+        // Disable 1ESPT schema and delete 1ESPT schema file if user is signed out
+        config.update('1ESPipelineTemplatesSchemaFile', false, vscode.ConfigurationTarget.Workspace); // disable 1ESPT schema
+        logger.log("1ESPT schema disabled as user is not signed in", 'SchemaDetection')
+        // show message to user that 1ESPT schema is disabled as user is not signed in
+        vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsUserNotSignedInMessage)
+
         return undefined;
     }
 
     // Create the global storage folder to guarantee that it exists.
     await vscode.workspace.fs.createDirectory(context.globalStorageUri);
+    
+    logger.log(`Retrieving schema for ${workspaceFolder.name}`, 'SchemaDetection');
+
+    // Try to fetch schema in the following order:
+    // 1. Cached 1ESPT schema
+    // 2. 1ESPT schema if user is signed in with microsoft account and has enabled 1ESPT schema
+    // 3. Cached Organization specific schema
+    // 4. Organization specific schema
+    const token = await session.credentials2.getToken();
+    const authHandler = azdev.getBearerHandler(token.accessToken);
+    const azureDevOpsClient = new azdev.WebApi(`https://dev.azure.com/${organizationName}`, authHandler);
+
+    if (vscode.workspace.getConfiguration('azure-pipelines', workspaceFolder).get<boolean>('1ESPipelineTemplatesSchemaFile', false)) {
```

Comment:
Optional: both getCached1ESPTSchema and get1ESPTSchema start out with `@microsoft.com` checks. I'm not sure if it's worth consolidating that here, outside of those functions - up to you.
(It would eliminate the need to pass in `session`, but at the same time it begins to blur the lines between who is responsible for what)